spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -63,97 +78,98 @@ protected AnalyzeTableSparkAction self() {
+    FileIO fileIO = operations.io();
+    String path = operations.metadataFileLocation(String.format("%s.stats", UUID.randomUUID()));
+    OutputFile outputFile = fileIO.newOutputFile(path);
+    GenericStatisticsFile statisticsFile;
+    try (PuffinWriter writer =
+        Puffin.write(outputFile).createdBy("Iceberg Analyze action").build()) {
+      blobs.forEach(writer::add);
+      writer.finish();
+      statisticsFile =
+          new GenericStatisticsFile(
+              snapshotId,
+              path,
+              writer.fileSize(),
+              writer.footerSize(),
+              writer.writtenBlobsMetadata().stream()
+                  .map(GenericBlobMetadata::from)
+                  .collect(ImmutableList.toImmutableList()));
+    }
+    table.updateStatistics().setStatistics(snapshotId, statisticsFile).commit();
   }
 
-  private AnalysisResult generateNDVAndCommit() {
-    try {
-      if (snapshotId == null) {
-        snapshotId = table.currentSnapshot().snapshotId();
-      }
-
-      StatisticsFile statisticsFile =
-          NDVSketchGenerator.generateNDV(
-              spark(), table, snapshotId, columns.toArray(new String[0]));
-      table.updateStatistics().setStatistics(snapshotId, statisticsFile).commit();
-      return ImmutableAnalyzeTable.AnalysisResult.builder()
-          .type(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1)
-          .build();
-    } catch (IOException ioe) {
-      List<String> errors = Lists.newArrayList();
-      errors.add(ioe.getMessage());
-      return ImmutableAnalyzeTable.AnalysisResult.builder()
-          .type(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1)
-          .addAllErrors(errors)
-          .build();
-    }
+  private List<Blob> generateNDVBlobs() {
+    return NDVSketchGenerator.generateNDVSketchesAndBlobs(spark(), table, snapshotId, columns);
   }
 
   @Override
   public AnalyzeTable columns(String... columnNames) {
     Preconditions.checkArgument(
         columnNames != null && columnNames.length > 0, "Columns cannot be null/empty");
-    Set<String> columnsSet = Sets.newHashSet(Arrays.asList(columnNames));
-    Preconditions.checkArgument(
-        analyzableTypes(columnsSet),
-        "Cannot be applied to the given columns, since the column's type is not supported");
-    this.columns = columnsSet;
+    for (String columnName : columnNames) {
+      Types.NestedField field = table.schema().findField(columnName);
+      if (field == null) {
+        throw new ValidationException("No column with %s name in the table", columnName);
+      }
+    }
+    this.columns = ImmutableSet.copyOf(columnNames);
     return this;
   }
 
   @Override
-  public AnalyzeTable types(Set<String> statisticTypes) {
-    Preconditions.checkArgument(
-        Sets.newHashSet(StandardBlobTypes.blobTypes()).containsAll(statisticTypes),
-        "type not supported");
-    this.types = statisticTypes;
+  public AnalyzeTable blobTypes(Set<String> types) {
```

Comment:
nit: types -> blobTypes (just like in the interface method declaration)

Reply:
Renamed